### ISSUE TITLE
Ability to disable thumbnail generator

### ISSUE THREAD

**Issue by astandarduser** [f3d-app/f3d#349]
Windows Installer Extensions and Thumbnailer checkboxes are not working

Steps to reproduce:
- Clean Windows system
- Install F3D 1.2.1, uncheck both extensions and thumbnailer checkboxes
- open a folder with compatible data in
- thumbnailers kicks off and extensions are associated, this is unexpected

OP:

I do really like f3d as a viewer because it seems to open models faster than anything else I have tried along with supporting more formats, however due to the current bugs in regards to the thumbnail generating it would be good to have a setting to turn it off and be just the f3d logo. I did try installing with the thumbnail generating textbox disabled but it doesn't seem to change anything

**Comment by mwestphal** [comment]
we will take a look, in the meantime you can use the zip archive instead of the installer if you want.

**Comment by mwestphal** [comment]
I confirm that disabling the checkbox seems to have no effect.

**Comment by mwestphal** [comment]
Another workaround is to set the folder option in windows: "Always show icons, never show thumbnails"

BTW, which bugs makes you want to disable to thumbnails generation ? I know there are a few but I dont see anything critical.

**Comment by mwestphal** [comment]
Here is the right way to disable the thumbnailer on windows, it is documented in the README.md:

open command prompt
```
cd C:\Program Files\f3d\bin\
regsvr32 /u F3DShellExtension.dll
```

But this should not be necessary

**Comment by mwestphal** [comment]
@astandarduser I took the liberty to edit your message.

@jpouderoux : Any idea what is going on ? I tried looking at the NSIS code but I never worked with it.

**Comment by astandarduser** [comment]
Thank you so much for the cmd workaround, I'll give it a try tonight. Disabling thumbnails in windows all together wouldn't be practical in my case as I find thumbnails very useful. 

The bug I faced I already reported on here a few months back about heavy system resource use and I use the latest release which is from before reporting that so not sure if it was fixed. Adding an easy option to disable it within the program would be a great feature regardless in times where a new bug arises if any, since most software provides a similar option. I do thing having thumbnails for 3D models is one of the best features of this software so once it is ironed out I'll certainly be enabling thumbnail generation again

**Comment by mwestphal** [comment]
Right, this one: https://github.com/f3d-app/f3d/issues/300

Yes it was not fixed yet.

> Adding an easy option to disable it within the program would be a great feature regardless in times where a new bug arises if any, since most software provides a similar option.

The way thumbnails are currently setup on windows makes it hard. We plan to add thumbnaisl specific config file in the future, so a way to simply disabling them in this config file should be easy to add.

**Comment by jpouderoux** [comment]
I will check the issue when I get back from vacations in 2 weeks.
In the meanwhile, you can always unregister the shell extension manually by running the following command (in cmd, powershell or the Run shortcut): 
`regsvr32.exe  /s Path_to_f3d\bin\F3DShellExtension.dll`

**Comment by astandarduser** [comment]
> 

Awesome, I'm looking forward to that release. I did reference another software that does 3d thumbnails with very little recourse usage in windows so not sure if the same method they used is viable although it isn't made with the same programming language. As I've had that one generating thumbs currently but wanted to use f3d again as a viewer which is why I posted this question, will keep an eye on the releases and test them/report back findings in future. Thanks again for your help
